src/App.js

```diff
@@ -21,32 +21,52 @@ class App extends Component {
   }
 
   componentDidMount() {
-    fetchAllSongs()
-      .then(data => Promise.all(data))
+    fetchAllSongData()
       .then(data => {
-        this.setState({ songBook: data })
-        this.setState({renderedSongs: [...this.state.songBook]})
+        this.modifyData(data[0], data[1])
```

Comment:
What if we destructured the data here:

 const [songs, genres] = data
        this.modifyData(songs, genres)

Just for readability purposes

Reply:
you got it!